accelerate-llvm-native/src/Data/Array/Accelerate/LLVM/Native/Link/Runtime.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE CPP               #-}
+{-# LANGUAGE OverloadedStrings #-}
+-- |
+-- Module      : Data.Array.Accelerate.LLVM.Native.Link.Runtime
+-- Copyright   : [2022] The Accelerate Team
+-- License     : BSD3
+--
+-- Maintainer  : Trevor L. McDonell <trevor.mcdonell@gmail.com>
+-- Stability   : experimental
+-- Portability : non-portable (GHC extensions)
+--
+-- Utilities for linking object code to shared objects and loading those
+-- generated shared objects on Unix-like systems.
+--
+
+module Data.Array.Accelerate.LLVM.Native.Link.Runtime (
+
+  loadSharedObject
+
+) where
+
+import Data.Array.Accelerate.Error
+import Data.Array.Accelerate.Lifetime
+
+import Data.Array.Accelerate.LLVM.Native.Link.Object
+import qualified Data.Array.Accelerate.LLVM.Native.Debug            as Debug
+
+import Control.Monad
+import Data.ByteString.Short.Char8                                  ( ShortByteString )
+import Formatting
+import qualified Data.ByteString.Short.Char8                        as B8
+
+#if defined(mingw32_HOST_OS)
+#error "Dynamic linking not implemented yet"
+#else
+import System.Posix.DynamicLinker
+#endif
+
+
+-- Dynamic object loading
+-- ----------------------
+
+-- Load the shared object file and return pointers to the executable
+-- functions defined within
+--
+loadSharedObject :: HasCallStack => [ShortByteString] -> FilePath -> IO (FunctionTable, ObjectCode)
+loadSharedObject nms path = do
+  so      <- dlopen path []
```

Comment:
Argument are missing for `dlopen()`. You need at least `RTLD_LAZY` so the symbols are lazily loaded, and you also want `RTLD_LOCAL` so the symbols defined in the kernel are not leaked to the global namespace (`RTLD_LOCAL` is the default, but it never hurts to be specific):

https://github.com/robbert-vdh/accelerate-llvm/blob/760ecd3822d1fd103783725067d4eb0de124511a/accelerate-llvm-native/src/Data/Array/Accelerate/LLVM/Native/Link/Posix.hs#L89

Reply:
fixed